packages/mui-base/src/useList/useListItem.ts

```diff
@@ -101,12 +102,14 @@ export function useListItem<ItemValue>(
     tabIndex = highlighted ? 0 : -1;
   }
 
-  const getRootProps = <TOther extends EventHandlers = {}>(
-    otherHandlers: TOther = {} as TOther,
+  const getRootProps = <ExternalProps extends Record<string, any>>(
+    externalProps: ExternalProps = {} as ExternalProps,
   ) => ({
-    ...otherHandlers,
-    onClick: createHandleClick(otherHandlers),
-    onPointerOver: handlePointerOverEvents ? createHandlePointerOver(otherHandlers) : undefined,
+    ...externalProps,
+    onClick: createHandleClick(extractEventHandlers(externalProps)),
```

Comment:
Let's call `extractEventHandlers` just once and pass the result to `createHandleClick` and `createHandlePointerOver`